examples/trusted_database/module_0/rust/tests/integration_test.rs

```diff
@@ -0,0 +1,172 @@
+        <locked>false</locked>
+        <installDate>0</installDate>
+        <removalDate/>
+        <temporary>false</temporary>
+        <nbBikes>1</nbBikes>
+        <nbEmptyDocks>1</nbEmptyDocks>
+        <nbDocks>1</nbDocks>
+    </station>
+    <station>
+        <id>4</id>
+        <name>Opened station 2</name>
+        <terminalName>4</terminalName>
+        <lat>10.0</lat>
+        <long>0.0</long>
+        <installed>true</installed>
+        <locked>false</locked>
+        <installDate>0</installDate>
+        <removalDate/>
+        <temporary>false</temporary>
+        <nbBikes>1</nbBikes>
+        <nbEmptyDocks>1</nbEmptyDocks>
+        <nbDocks>1</nbDocks>
+    </station>
+</stations>"#;
+
+fn build_wasm() -> anyhow::Result<HashMap<String, Vec<u8>>> {
+    Ok(hashmap! {
+        MAIN_MODULE_NAME.to_owned() => oak_tests::compile_rust_wasm(MAIN_MODULE_MANIFEST, MAIN_MODULE_WASM_FILE_NAME).context("Couldn't compile main module")?,
+        HANDLER_MODULE_NAME.to_owned() => oak_tests::compile_rust_wasm(HANDLER_MODULE_MANIFEST, HANDLER_MODULE_WASM_FILE_NAME).context("Couldn't compile handler module")?,
+    })
+}
+
+async fn list_points_of_interest(
+    client: &mut TrustedDatabaseClient<tonic::transport::Channel>,
+    latitude: f32,
+    longitude: f32,
+) -> Result<tonic::Response<ListPointsOfInterestResponse>, tonic::Status> {
+    let request = ListPointsOfInterestRequest {
+        location: Some(Location {
+            latitude,
+            longitude,
+        }),
+    };
+    client.list_points_of_interest(request).await
+}
+
+#[tokio::test(core_threads = 2)]
+async fn test_trusted_database() {
+    env_logger::init();
+
+    // Send test database as a start-of-day config map.
+    let config_map = ConfigMap {
+        items: hashmap! {"database".to_string() => XML_DATABASE.as_bytes().to_vec()},
+    };
+    let wasm_modules = build_wasm().expect("Couldn't build wasm modules");
+    let config = oak_tests::runtime_config_wasm(
+        wasm_modules,
+        MAIN_MODULE_NAME,
+        MAIN_ENTRYPOINT_NAME,
+        config_map,
+    );
+    let runtime =
+        oak_runtime::configure_and_run(config).expect("Couldn't configure runtime with test wasm");
+
+    let (channel, interceptor) = oak_tests::channel_and_interceptor().await;
+    let mut client = TrustedDatabaseClient::with_interceptor(channel, interceptor);
+
+    // Test nearest point of interest.
+    let result = list_points_of_interest(&mut client, 4.0, -10.0).await;
+    assert_matches!(result, Ok(_));
+    let got = result.unwrap().into_inner();
+    let want = ListPointsOfInterestResponse {
+        point_of_interest: Some(PointOfInterest {
+            name: "Opened station 1".to_string(),
+            location: Some(Location {
+                latitude: 0.0,
+                longitude: 0.0,
+            }),
+        }),
+    };
+    assert_eq!(got, want);
+
+    let result = list_points_of_interest(&mut client, 9.0, -10.0).await;
```

Comment:
Is this second request testing something in addition to the first request? If so, please add a comment to explain what it is testing. Otherwise, I guess one request is enough to show that the correct point of interest is retrieved.

Reply:
I agree with @ipetr0v , I think 2 examples is better here, to corroborate (although not prove, of course) the idea that at least it is not always returning the first entry or something like that.

Reply:
Makes sense. Would be nice to have a comment line to mention this :)

Reply:
Done